cadquery/occ_impl/importers/assembly.py

```diff
@@ -0,0 +1,216 @@
+from OCP.TDF import TDF_Label, TDF_LabelSequence, TDF_AttributeIterator
+from OCP.IFSelect import IFSelect_RetDone
+from OCP.TDocStd import TDocStd_Document
+from OCP.TDataStd import TDataStd_Name
+from OCP.TNaming import TNaming_NamedShape
+from OCP.STEPCAFControl import STEPCAFControl_Reader
+from OCP.XCAFDoc import XCAFDoc_ColorSurf, XCAFDoc_DocumentTool, XCAFDoc_GraphNode
+
+import cadquery as cq
+from ..assembly import AssemblyProtocol
+
+
+def importStep(assy: AssemblyProtocol, path: str):
+    """
+    Import a step file into an assembly.
+
+    :param assy: An Assembly object that will be packed with the contents of the STEP file.
+    :param path: Path and filename to the STEP file to read.
+
+    :return: None
+    """
+
+    def _process_label(lbl: TDF_Label):
+        """
+        Recursive method to process the assembly in a top-down manner.
+        """
+
+        # Instantiate the new assembly
+        new_assy = cq.Assembly()
+
+        # Look for components
+        comp_labels = TDF_LabelSequence()
+        shape_tool.GetComponents_s(lbl, comp_labels)
+
+        for i in range(comp_labels.Length()):
+            comp_label = comp_labels.Value(i + 1)
+
+            # Get the location of the component label
+            loc = shape_tool.GetLocation_s(comp_label)
+            cq_loc = cq.Location(loc) if loc else None
+
+            if shape_tool.IsReference_s(comp_label):
+                ref_label = TDF_Label()
+                shape_tool.GetReferredShape_s(comp_label, ref_label)
+
+                # Find the name of this referenced part
+                ref_name_attr = TDataStd_Name()
+                if ref_label.FindAttribute(TDataStd_Name.GetID_s(), ref_name_attr):
+                    ref_name = str(ref_name_attr.Get().ToExtString())
+
+                if shape_tool.IsAssembly_s(ref_label):
+                    # Recursively process subassemblies
+                    sub_assy = _process_label(ref_label)
+
+                    # Add the appropriate attributes to the subassembly
+                    new_assy.add(sub_assy, name=f"{ref_name}", loc=cq_loc)
+                elif shape_tool.IsSimpleShape_s(ref_label):
+                    # A single shape needs to be added to the assembly
+                    final_shape = shape_tool.GetShape_s(ref_label)
+                    cq_shape = cq.Shape.cast(final_shape)
+
+                    # Find the subshape color, if there is one set for this shape
+                    color = Quantity_ColorRGBA()
+                    # Extract the color, if present on the shape
+                    if color_tool.GetColor(final_shape, XCAFDoc_ColorSurf, color):
+                        rgb = color.GetRGB()
+                        cq_color = cq.Color(
+                            rgb.Red(), rgb.Green(), rgb.Blue(), color.Alpha()
+                        )
+                    else:
+                        cq_color = None
+
+                    new_assy.add(
```

Comment:
So probably some logic should be added here to handle the `name` / `name_part` pairs. E.g. something like:
```
if ref_name.endswith('_part') and ref_name.startswith(parent_name):
    new_assy[parent_name].obj = cq_shape
else:
    ...
```
CC @lorenzncode